src/components/common/PeopleAndRoles/PeopleAndRoles.vue

```diff
@@ -456,9 +527,15 @@ export default class PeopleAndRoles extends Mixins(CommonMixin, DateMixin, OrgPe
     } else {
       // get ID of edited person to undo
       const id = person?.officer?.id
-
-      // get a copy of original person from original IA
-      const thisPerson = (id !== undefined) && cloneDeep(this.originalParties.find(x => +x.officer.id === +id))
+      const regexExp = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/gi
+
+      let thisPerson
+      if (id !== undefined && regexExp.test(id)) {
+        thisPerson = cloneDeep(this.originalParties.find(x => x.officer.id === id))
+      } else {
+        // get a copy of original person from original IA
+        thisPerson = cloneDeep(this.originalParties.find(x => +x.officer.id === +id))
+      }
 
```

Comment:
We can't reliably convert a UUID to a JavaScript/TypeScript number. The number type is a floating-point type and becomes potentially imprecise. Hence, compared as above (String)

Reply:
Good thinking!

Reply:
Why are you looking at the UUID like this? I don't remember other code where we need to check anything more than "does the id exist?".

On line 533, can't you simply do `if (!id)` ? (If it helps, you can check `isNaN(id)` to see whether it's a number or not.)

Reply:
isNaN(id) works. Thanks Sev! New learning for me :)

Reply:
Great! You can resolve this conversation if you want.